### ISSUE TITLE
Change Mujoco Model Parameters online

### ISSUE THREAD

**Issue by MankowitzProjects** [openai/mujoco-py#33]
Hi there,

I am not sure if this is the correct forum for this question but I would like to ask whether it is possible, using your python wrappers, to change the Mujoco model parameters dynamically between time steps. For example, in inverted pendulum, can I change the mass or length of the pendulum at each time step?

Thanks!

**Comment by tlbtlbtlb** [comment]
There's no high-level API through Mujoco-py. It might be possible using Mujoco's C API. See http://mujoco.org/book/programming.html.

**Comment by MankowitzProjects** [comment]
Yeah it is definitely possible using Mujoco's C API. The only issue is that I will need to recompute parameters that are saved in mjModel on the fly, by calling mj_setConst on the C side. Do you see a way for me to modify the python code to perform this call on the fly from the python side?

**Comment by tlbtlbtlb** [comment]
Uncomment the definition of mj_setConst here: https://github.com/openai/mujoco-py/blob/master/mujoco_py/mjlib.py#L77 and you should be able to call it.

**Comment by MankowitzProjects** [comment]
Great thanks Trevor!

**Comment by Kiwoo** [comment]
Hi,

Could you explain how you solved this problem?
I am not sure how to use Mujoco's C Api with openai Gym Wrapper.

Thank you.

**Comment by andrewliao11** [comment]
hi @MankowitzProjects, 
I wonder how you change the mass of certain object in mujoco. 
I just umcomment [this line](https://github.com/openai/mujoco-py/blob/master/mujoco_py/mjlib.py#L77), and then I'm at lost :(

**Comment by andrewliao11** [comment]
Thanks I solved it

**Comment by MankowitzProjects** [comment]
Hi Andrew,

I hadn't gotten around to implementing the model parameter changes as yet.
I am glad you solved it!

Cheers,
Dan

**Comment by andrewliao11** [comment]
hi @tlbtlbtlb 
I'm still new to mujoco. Can you give us an example use of mjlib.mj_setConst?? 
It seems that the argument should be [POINTER(MJMODEL), POINTER(MJDATA), c_int]
Whereas,  I still don't know how to use it

**Comment by jtoyama4** [comment]
Hi,
I am also struggling to change the model parameters such as arm's length at each episode.
Could anyone give me an example how to do it?
I am not sure what to do with mjlib.mj_setConst.
Thanks,

**Comment by jtoyama4** [comment]
I just found same question in mujoco forum and it answers more specifically.

http://www.mujoco.org/forum/index.php?threads/can-the-parameters-of-a-model-be-changed-on-the-fly.3354/

**Comment by Joll123** [comment]
hi @MankowitzProjects,
how to use mj_setConst on  the python side.
thanks
